### ISSUE TITLE
Unable to process `count` query of undefined

### ISSUE THREAD

**Issue by zomars** [prisma/studio#890]
When trying to use Studio this error happens:

![image](https://user-images.githubusercontent.com/3504472/159522873-0fa0b31d-be4a-439c-a625-c2ae00501f2b.png)

Since we're open source we can share our prisma schema without problems:

https://github.com/calcom/cal.com/blob/main/packages/prisma/schema.prisma

We can run queries just fine, it just crashes when trying to open up Studio.

_Originally posted by @zomars in https://github.com/prisma/studio/issues/871#issuecomment-1075347171_

**Comment by janpio** [comment]
> We can run queries just fine

What exactly does this refer to? The Query Console of Prisma Data Platform?

Can you maybe share your project name and URL on cloud.prisma.io?

**Comment by zomars** [comment]
> > We can run queries just fine
> 
> What exactly does this refer to? The Query Console of Prisma Data Platform?

The Data Browser specifically is the one failing

> Can you maybe share your project name and URL on cloud.prisma.io?

Sure thing, it is `https://cloud.prisma.io/zomars/calcom/staging/databrowser`

**Comment by zomars** [comment]
This only seems to happen when provisioning my own DB url, generating one from Heroku doesn't have this problem.

**Comment by janpio** [comment]
Interesting. Where do you host that database? Anything special about it or the user you use to connect to it? 
Do you have any output in the developer console when opening the data browser? Maybe the specific "query" that fails...
Are you sure the database is fully migrated to the state of the schema file that the data browser is using (via the repo that you connected)?

**Comment by MaxMusing** [comment]
Running into this issue too.

The full error message in the browser console says:
> The provided database string is invalid. Error parsing connection string: empty host in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.

The way we're setting our database URL is a little funky because we're using [Doppler](https://www.doppler.com) to manage secrets. Our `schema.prisma` looks like this:
```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

And our `.env` file looks like this:
```
DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}
```

Where those environment variables are defined in Doppler.


If I hard-code the `url` in `schema.prisma` to be a fully-defined database URL, then I'm able to load Prisma Studio.

**Comment by janpio** [comment]
Max, I think that sounds like an unrelated problem that just surfaces the same way. Can you please open a new issue?
(Are you trying to start Studio locally, or are you on cloud.prisma.io in the Data Browser as Omar?)

**Comment by oreqizer** [comment]
@zomars are you by any chance using Safari? I had this error, and it only appears in Safari. works ok in Chrome

**Comment by zomars** [comment]
> @zomars are you by any chance using Safari? I had this error, and it only appears in Safari. works ok in Chrome

Yes I'm using Safari. Didn't mention it since I don't think it should affect, but seems I'm wrong.

**Comment by matthewp672261** [comment]
> @zomars are you by any chance using Safari? I had this error, and it only appears in Safari. works ok in Chrome

switching from Safari to chrome fixed this issue for me

**Comment by oreqizer** [comment]
I found this is only happening if my database is misaligned with my Prisma schema — are you sure yours are in sync?

**Comment by neoromantic** [comment]
Still happening. Confirming that this happens in Safari, but not in Chrome. Seems like a critical bug to me.

**Comment by petradonka** [comment]
@neoromantic, are you also using the Data Browser in Prisma Data Platform (if yes, could you share the owner and project name)?
What database are you using? Do you have a large number of relationships?

**Comment by zomars** [comment]
This is still happening on latest prisma BTW: 
<img width="474" alt="image" src="https://user-images.githubusercontent.com/3504472/184950079-59905bde-306b-405c-ac0c-f3911e2f99cc.png">

**Comment by jeffdrumgod** [comment]
same here, but using a custom folder for Prisma, for MongoDB.
The nodejs application works fine, but Prisma Studio show this error when it start Prisma studio.

**Comment by petradonka** [comment]
Thanks for the extra info! Do you have a relatively large number of relationships?

**Comment by mcgri** [comment]
npx prisma db push <- solved safari issue for me too.

**Comment by argcast** [comment]
This might happen while trying to load non existing tables on current database but existing on Prisma's schema. As mentioned above, make sure to `npx prisma db push` and then re-launch studio. Solved it for me too.

**Comment by cerpow** [comment]
Happened to me if you set the mongo url in env() function 

`url = env("DATABASE_URL")`

should be :
`url = "DATABASE_URL"`

**Comment by XDCoder3289** [comment]
> @zomars are you by any chance using Safari? I had this error, and it only appears in Safari. works ok in Chrome

That is weird. Didn't know this could have been a problem

**Comment by bab55z** [comment]
Maybe you should try using [`dotenv`](https://www.npmjs.com/package/dotenv-cli) cli to provide correct environment variables to Studio
`yarn global add dotenv-cli`
`dotenv -e .env.local -- npx prisma studio`

**Comment by do-kevin** [comment]
Can confirm.

`npx prisma db push` fixed it for Safari. Chrome already worked.

**Comment by jingidy** [comment]
If anyone's encountering this with supabase and pgbouncer, and using different ports for pooling and non-pooling, try changing the port numbers, that fixed it for me. Now I just need to check if the port actually pools or if my code reversed the two.

**Comment by niloy-farhan** [comment]
> same here, but using a custom folder for Prisma, for MongoDB. The nodejs application works fine, but Prisma Studio show this error when it start Prisma studio.
> 
> ✌️ **Note**: This error occurs only on Safari. Opening the same URL/project in Chrome, error disappear

drum bro, Thanks

**Comment by nwazuo** [comment]
> I found this is only happening if my database is misaligned with my Prisma schema — are you sure yours are in sync?

This is the case with me too. Running `prisma db push` fixes it🚀.

**Comment by Isypzenko** [comment]
I've fixed that. My .env file had gone somewhere..., I just add url provider in schema.prisma file and it worked

**Comment by igalklebanov** [comment]
Hey :wave:

Prisma ORM 7 introduces a brand new standalone [Studio](https://www.prisma.io/docs/orm/tools/prisma-studio), built from the ground up to work everywhere, with or without Prisma ORM. Consequently, all "old Studio" issues are being closed as there won't be any further work on it. :mega:

The new Studio is also available in the Prisma VS Code extension and in the Prisma Console application. :electric_plug:

Help shape, improve and fix the new Studio with us in this repository by submitting bug reports and feature requests. :construction_worker:
